### PR TITLE
FE-1085 - Update Hibana Matchbox to 4.0.0-hibana.24 and Matchbox Icons to 1.3.0

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-1085](https://jira.int.messagesystems.com/browse/FE-1085)

### What Changed
- Installs the Hibana version of Matchbox, version 23
- Updates the Matchbox icons package to `1.3.0`
- Installs `styled-components` - now a peer dependency of Matchbox
- Updated a few Cypress tests that were failing as a result of this upgrade - Modals more frequently require the `force: true` to click on elements within them
- Removes background on the `SortLabel` component - a bug introduced by a new version of Chrome
- Removed custom positioning within the `ShareModal` component that was fixed with the Matchbox upgrade
 - Used new `disableResponsiveBehavior` prop on the templates tabs to prevent overflowing from rendering a popover
- Removed some hacky CSS that Matchbox fixed with the upgrade

### How To Test
- Verify unit/integration tests continue to pass
- Take a look at release notes to check for potentially breaking changes - they are linked via the Jira ticket
- Click around for anything funky!

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

**Comment by nicklemmon** [comment]
This is on hold for a bit - needs an update to the tabs component before proceeding.

File: src/pages/templates/components/EditSectionHibana.module.scss [hunk 1]
```diff
@@ -14,6 +14,7 @@
   // TODO: Need a way to disable the bottom border on the <Tabs/> component
   > div {
     border-bottom: 0;
+    overflow: visible;
```
Comment (nicklemmon) [inline src/pages/templates/components/EditSectionHibana.module.scss:17]: This was needed to address a bug with Matchbox. Jon is working on addressing:
https://github.com/SparkPost/matchbox/pull/435

File: src/pages/templates/components/editorActions/ActionsHibana.module.scss [hunk 2]
```diff
@@ -40,10 +40,6 @@
   margin-left: spacing(500);
 }
 
-.ActionsPopoverTrigger {
```
Comment (nicklemmon) [inline src/pages/templates/components/editorActions/ActionsHibana.module.scss:43]: Matchbox update fixed bug that was causing this one, so removed the hacky CSS

File: src/pages/templates/components/tests/EditSection.test.js [hunk 3]
```diff
@@ -25,11 +25,6 @@ describe('EditSection', () => {
     return shallow(<EditSection />);
   };
 
-  it('renders tabs and section', () => {
-    const wrapper = subject({ currentTabIndex: 0 });
-    expect(wrapper).toMatchSnapshot();
-  });
```
Comment (nicklemmon) [inline src/pages/templates/components/tests/EditSection.test.js:31]: Not a useful test. Just removed it.

File: src/pages/templates/components/editorActions/ActionsHibana.module.scss (hunk 2 above)
Comment (kchu1337) [inline src/pages/templates/components/editorActions/ActionsHibana.module.scss:43]: Could you also delete the code where it references this style in `src/pages/templates/components/editorActions/PublishedModeActions.js`?

**Pull request merged**